hudi-client/src/main/java/org/apache/hudi/HoodieWriteClient.java

```diff
@@ -1171,9 +1175,13 @@ private void rollbackInflightCommits() {
     }
   }
 
-  private HoodieTable getTableAndInitCtx() {
+  private HoodieTable getTableAndInitCtx(String operationType) {
+    HoodieTableMetaClient metaClient = createMetaClient(true);
+    if (operationType.equalsIgnoreCase(DELETE_STR)) {
+      addSchemaToConfig(metaClient);
```

Comment:
@bvaradar : Do you think we should check if config doesn't have schema set and then add it need be. Bcoz, if someone instantiates the client, does a bunch of inserts and updates and then deletes, config will have proper schema set. The problem arises only when someone instantiates the client and issues delete(which may not have schema set)